File: python/k8s_full_stack_cdk/networking_stack.py
from aws_cdk import (
    # Duration,
    NestedStack,
    aws_ec2 as ec2,
)
from constructs import Construct
from helper import config

class NetworkingStack(NestedStack):

    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        conf = config.Config(self.node.try_get_context('environment'))

        self.vpc = ec2.Vpc(self, "k8s-sample-vpc",
            ip_addresses=ec2.IpAddresses.cidr(conf.get('cidr')),
            subnet_configuration=[
                ec2.SubnetConfiguration(
                    name = 'public',
                    subnet_type = ec2.SubnetType.PUBLIC,
                    cidr_mask = 28
                ),
                ec2.SubnetConfiguration(
                    name = 'eks',
                    subnet_type = ec2.SubnetType.PRIVATE_WITH_EGRESS,
                    cidr_mask = 26
                )
            ],
        )

File: go/stacks/networking-stack.go
package stacks

import (
	"github.com/aws/aws-cdk-go/awscdk/v2"
	"github.com/aws/aws-cdk-go/awscdk/v2/awsec2"
	"github.com/aws/constructs-go/constructs/v10"
	// https://docs.aws.amazon.com/sdk-for-go/api/aws/
	"github.com/aws/aws-sdk-go-v2/aws"

	"k8s-full-stack-cdk-go/helper"
)

type NetworkingNestedStackProps struct {
	awscdk.NestedStackProps
}

func NetworkingStack(scope constructs.Construct, id string, config helper.Conf, props *NetworkingNestedStackProps) (awscdk.NestedStack, awsec2.IVpc) {
	var nsprops awscdk.NestedStackProps
	if props != nil {
		nsprops = props.NestedStackProps
	}
	stack := awscdk.NewNestedStack(scope, &id, &nsprops)

	vpc := awsec2.NewVpc(stack, aws.String("k8s-sample-vpc"), 
		&awsec2.VpcProps{
			IpAddresses: awsec2.IpAddresses_Cidr(aws.String(config.CIDR)),
			SubnetConfiguration: &[]*awsec2.SubnetConfiguration{
				{
					CidrMask: aws.Float64(28),
					SubnetType: awsec2.SubnetType_PUBLIC,
					Name: aws.String("public"),
				},
				{
					CidrMask: aws.Float64(26),
					SubnetType: awsec2.SubnetType_PRIVATE_WITH_EGRESS,
					Name: aws.String("eks"),
				},
			},
		},
	)

	return stack, vpc
}